dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java

```diff
@@ -35,33 +40,65 @@
 
     protected RestCallback<?> callback;
     protected DelegatingDispatchRequest delegatingDispatchRequest;
+    protected SuccessCallback successCallback;
+    protected ResponseCallback alwaysCallback;
+    protected FailureCallback failureCallback;
 
     protected AbstractResourceDelegate(RestDispatch dispatcher) {
         this.dispatcher = dispatcher;
     }
 
     @Override
-    public ResourceDelegate<T> withDelegatingDispatchRequest(DelegatingDispatchRequest delegatingDispatchRequest) {
-        AbstractResourceDelegate<T> delegate = createCopy();
+    public ResourceDelegate<R> withDelegatingDispatchRequest(DelegatingDispatchRequest delegatingDispatchRequest) {
+        AbstractResourceDelegate<R> delegate = createCopy();
         delegate.delegatingDispatchRequest = delegatingDispatchRequest;
 
         return delegate;
     }
 
     @Override
-    public T withoutCallback() {
+    public R withoutCallback() {
```

Comment:
I think we can safely remove that method. wrapCallback will have the same effect if nothing is specified.